src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Tree/Visitors/ExpressionTreeRewriter.cs

```diff
@@ -46,19 +46,19 @@ protected override EXPR Dispatch(EXPR expr)
 
         /////////////////////////////////////////////////////////////////////////////////
         // Statement types.
-        protected override EXPR VisitASSIGNMENT(EXPRASSIGNMENT assignment)
+        protected override Expr VisitASSIGNMENT(ExprAssignment assignment)
```

Comment:
There are more similar changes to be done, but I think at 2000+ lines just from doing Expr an types derived from it I'll leave others for later.